pkg/services/sqlstore/dashboard_version.go

```diff
@@ -58,3 +64,72 @@ func GetDashboardVersions(query *m.GetDashboardVersionsQuery) error {
 	}
 	return nil
 }
+
+func DeleteExpiredVersions(cmd *m.DeleteExpiredVersionsCommand) error {
+	return inTransaction(func(sess *DBSession) error {
+		var expiredCount int64 = 0
+		var versions []DashboardVersionExp
+
+		// Don't clean up if user set versions_to_keep to 2147483647 (MaxInt32)
+		if versionsToKeep := setting.DashboardVersionsToKeep; versionsToKeep < math.MaxInt32 {
+			// Get dashboard ids to clean up
+			affectedDashboardsQuery := fmt.Sprintf(`SELECT dashboard_id FROM dashboard_version
+				GROUP BY dashboard_id HAVING COUNT(dashboard_version.id)>%d`, versionsToKeep)
```

Comment:
%d in a SQL query opens up SQL injection possibilities. Use `> ?` and send in versionToKeep as a number